### PR TITLE
Some Style Changes

### PR DESCRIPTION
**Pull request by Sharkmare**
Primarily just some stylistic adjustments.
Not sure if youd want those or not but making this PR just as an option

### CODE REVIEW

File: ColorMyLogiXNodes/ColorMyLogiXNodes.cs [hunk 1]
```diff
@@ -256,356 +257,389 @@ private enum InputNodeOverrideEnum
...
 				{
-					Debug("nodeInfoList Size before clear: " + nodeInfoSet.Count.ToString());
-					NodeInfoListClear();
-					Debug("Cleared nodeInfoList. New size: " + nodeInfoSet.Count.ToString());
+					ClearSyncRefTargetMap();
+					ClearNodeInfoSet();
 				}
 
-				if (configChangedEvent.Key == USE_AUTO_RANDOM_COLOR_CHANGE && Config.GetValue(USE_AUTO_RANDOM_COLOR_CHANGE))
+				if (useAutoRandomColorChange)
```
Comment (Nytra) [inline ColorMyLogiXNodes/ColorMyLogiXNodes.cs]: This should still be checking if `configChangedEvent.Key == USE_AUTO_RANDOM_COLOR_CHANGE`

File: ColorMyLogiXNodes/ColorMyLogiXNodes.cs [hunk 2]
```diff
@@ -256,356 +257,389 @@ private enum InputNodeOverrideEnum
...
-					syncRefTargetMap.Clear();
-					Debug("Cleared syncRefTargetMap. New size: " + syncRefTargetMap.Count.ToString());
-				}
+				bool modEnabled = Config.GetValue(MOD_ENABLED);
+				bool autoUpdateRefAndDriverNodes = Config.GetValue(AUTO_UPDATE_REF_AND_DRIVER_NODES);
+				bool updateNodesOnConfigChanged = Config.GetValue(UPDATE_NODES_ON_CONFIG_CHANGED);
+				bool useAutoRandomColorChange = Config.GetValue(USE_AUTO_RANDOM_COLOR_CHANGE);
 
-				if ((configChangedEvent.Key == MOD_ENABLED && !Config.GetValue(MOD_ENABLED)) || ((configChangedEvent.Key == UPDATE_NODES_ON_CONFIG_CHANGED || configChangedEvent.Key == USE_AUTO_RANDOM_COLOR_CHANGE && (!Config.GetValue(UPDATE_NODES_ON_CONFIG_CHANGED)) && !Config.GetValue(USE_AUTO_RANDOM_COLOR_CHANGE))))
+				if (!modEnabled || !autoUpdateRefAndDriverNodes)
```
Comment (Nytra) [inline ColorMyLogiXNodes/ColorMyLogiXNodes.cs]: Should be checking configChangedEvent.Key here

File: ColorMyLogiXNodes/ColorMyLogiXNodes.cs (hunk 1 above)
Comment (Sharkmare) [reply ColorMyLogiXNodes/ColorMyLogiXNodes.cs]: Added the check back

File: ColorMyLogiXNodes/ColorMyLogiXNodes.cs (hunk 2 above)
Comment (Sharkmare) [reply ColorMyLogiXNodes/ColorMyLogiXNodes.cs]: Added the check back

**Comment by Sharkmare** [comment]
Readded the keys i missed

**Comment by Nytra** [comment]
Some of these changes look good. Others I question. I will probably go through and cherry-pick ones that I like at some point.

**Comment by Sharkmare** [comment]
> Some of these changes look good. Others I question. I will probably go through and cherry-pick ones that I like at some point.

Curious about which ones you question, but beyond my curiosity that sounds good yeah, glad you found some of it useful.

File: ColorMyLogiXNodes/Colors.cs [hunk 3]
```diff
@@ -7,102 +7,66 @@ namespace ColorMyLogixNodes
...
+                case ColorModelEnum.RGB:
+                    color colorRGB = new color(Config.GetValue(NODE_COLOR));
+                    val = GetColorChannelValue(colorRGB.r, colorRGB.g, colorRGB.b, index);
+                    break;
+            }
+            return val;
+        }
+        private static float GetColorChannelValue(float channel1, float channel2, float channel3, int index)
+        {
+            return index switch
```
Comment (Nytra) [inline ColorMyLogiXNodes/Colors.cs:51]: What is this doing?

File: ColorMyLogiXNodes/Colors.cs [hunk 4]
```diff
@@ -295,19 +213,13 @@ private static color GetTextColor(color bg)
...
-			if (c.g > 1) c = c.SetG(1f);
-			if (c.g < 0) c = c.SetG(0f);
-
-			if (c.b > 1) c = c.SetB(1f);
-			if (c.b < 0) c = c.SetB(0f);
-
-			if (c.a > 1) c = c.SetA(1f);
-			if (c.a < 0) c = c.SetA(0f);
-		}
+            // Clamp color to min 0 and max 1 (no negative or emissive colors allowed)
```
Comment (Nytra) [inline ColorMyLogiXNodes/Colors.cs:216]: This is good

File: ColorMyLogiXNodes/Threads.cs [hunk 5]
```diff
@@ -10,124 +10,160 @@ namespace ColorMyLogixNodes
...
+                    continue;
+                }
+            }
+        }
+        private static int Clamp(int value, int minValue, int maxValue)
+        {
+            return Math.Min(Math.Max(value, minValue), maxValue);
+        }
+
+        private static bool IsNodeInvalid(NodeInfo nodeInfo)
```
Comment (Nytra) [inline ColorMyLogiXNodes/Threads.cs:130]: Also good

File: ColorMyLogiXNodes/ColorMyLogiXNodes.cs (hunk 2 above)
Comment (Nytra) [reply ColorMyLogiXNodes/ColorMyLogiXNodes.cs]: It is still missing some of the checks from before. I actually don't think this behaves the same as my original code.

**Comment by Nytra** [comment]
Added some of these changes in v1.2.1 https://github.com/Nytra/NeosColorMyLogiX/releases/tag/v1.2.1

**Comment by Sharkmare** [comment]
> Added some of these changes in v1.2.1 https://github.com/Nytra/NeosColorMyLogiX/releases/tag/v1.2.1

**Pull request closed without merging**